### PR TITLE
add a standard .rubocop.yml file

### PR DESCRIPTION
**Pull request by gergnz**
Requesting to add a standard rubocop.yml file to our style guide.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3644,6 +3644,8 @@ checker based on this style guide. RuboCop already covers a
 significant portion of the Guide, supports both MRI 1.9 and MRI 2.0
 and has good Emacs integration.
 
+Bulletproof's standard [.rubocop.yml](https://github.com/bulletproofnetworks/ruby-style-guide/blob/master/.gitattributes) is available in this repo.
```
Comment (nerdyness) [inline README.md]: .gitattributes?

**Comment by nerdyness** [comment]
LGTM! Awesome idea :+1:

**Comment by nerdyness** [comment]
@johnf, this shouldn't cause any issues upstream, right?

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Rocketships!!!!
+HashSyntax:
+  EnforcedStyle: hash_rockets
+
+# Enforce spaces within Braces
+Style/SpaceInsideHashLiteralBraces:
+  EnforcedStyle: space
+
+# Numbers without underscores, please.
+Style/NumericLiterals:
```
Comment (johnf) [inline .rubocop.yml]: I disagree with the one. The underscores make big numbers so much more readable

**Comment by johnf** [comment]
Is it worth adding a ```rubocop.yaml``` that symlinks to ```.rubocop.yaml``` so it's more obvious when you check it out?

**Comment by nerdyness** [comment]
> Is it worth adding a rubocop.yaml that symlinks to .rubocop.yaml so it's more obvious when you check it out?

I don't see any downside to that. Thanks for your input, @johnf. @gergnz I'll leave the honours of merging && deleting the branch to you :wink:

File: .rubocop.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  CountComments: false
+  Max: 150
+
+# Allow sightly longer lines
+Metrics/LineLength:
+  Max: 120
+
+# Allow slightly longer methods
+Metrics/MethodLength:
+  Max: 50
```
Comment (jcroteau) [inline .rubocop.yml]: 50 lines for a method??? :-1:  :smile:

File: .rubocop.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Rocketships!!!!
+HashSyntax:
+  EnforcedStyle: hash_rockets
```
Comment (jcroteau) [inline .rubocop.yml]: :-1:  x :100:

Comment (nerdyness) [inline .rubocop.yml]: :holywar: :wink:

**Comment by nerdyness** [comment]
@jcroteau you make some interesting suggestions. Is this something for the standards committee to talk about? I don't want to kick off another holy war like we did in Melbourne where we discussed this and came to the conclusions we have now in that yml file.

What are your suggestions? Obviously Ruby19 syntax for hash symbols but in terms of line/method/class/etc length?

File: .rubocop.yml (hunk 4 above)
Comment (johnf) [inline .rubocop.yml]: I'm going to pull out my CTO hat here. We've had this discussions many times and we even made a decision once.

Hashrockets it is

**Comment by jcroteau** [comment]
@nerdyness  I know the hash syntax is not a battle worth fighting, I was just making my voice heard (or rather thumbs down seen).

However, my method length concern is a bit more serious.  I think well-written methods should be 10-15 lines MAX, and even that is getting long. Certainly exceptions can be made in rare cases, but that's where the rubocop enable/disable magic comments come in.

**Comment by nerdyness** [comment]
> I think well-written methods should be 10-15 lines MAX, and even that is getting long. Certainly exceptions can be made in rare cases, but that's where the rubocop enable/disable magic comments come in.

I agree with you there, @jcroteau. I dunno how this yaml file came about and merely started using it. Some code I've seen in the wild has stupidly long methods and I know Lindsay would have ripped my head off for it in the code review :smile:

Somehow the code got out there and I'm guessing we're seeing the after-effects of that in this `.rubocop.yml`. I'd be happy to change that to 15 lines with the option of making exceptions in some cases.

thoughts everybody?

**Comment by gergnz** [comment]
@nerdyness I was surprised that this file had 50, but it helped me get rubocop to pass (hehe) but I guess that stops me from thinking how to write better code.  Lets make 15.

**Comment by jcroteau** [comment]
:+1: :+1: :+1:

**Pull request merged**